src/Cardano/PoolActions.hs

```diff
@@ -0,0 +1,112 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cardano.PoolActions where
+
+import           Ledger.Address
+import           Cardano.Models
+import           Cardano.Utils
+import           ErgoDex.Contracts.Pool
+import           ErgoDex.Contracts.Types
+import           Data.Either.Combinators         (maybeToRight)
+import           ErgoDex.OffChain
+import           ErgoDex.Types
+import           Plutus.V1.Ledger.Value
+import           Prelude                         ( String
+                                                 , Either(..)
+                                                 , fst
+                                                 , uncurry
+                                                 , (<>)
+                                                 , (.)
+                                                 , ($)
+                                                 , (==)
+                                                 , mempty
+                                                 , return
+                                                 , show)
+import qualified PlutusTx.AssocMap               as Map
+import           Ledger.Scripts
+import qualified Ledger.Typed.Scripts.Validators as Validators
+import qualified PlutusTx
+import           Data.List
+import           Data.Maybe
+
+newtype TxCandidateCreationErr = TxCandidateCreationErr { errMsg :: String }
+
+coinNotFoundError :: Coin a -> TxCandidateCreationErr
+coinNotFoundError coin =
+  TxCandidateCreationErr ( "Coin " ++ show coin ++ " not presented" )
+
+createPool :: PoolParams -> [(FullTxOut, Maybe Redeemer)] -> Address -> Either TxCandidateCreationErr TxCandidate
```

Comment:
Why not using FullTxIn here? `cast2Input` is not the responsibility of `createPool`.